packages/flutter_tools/test/general.shard/asset_bundle_test.dart

```diff
@@ -171,14 +171,9 @@ flutter:
         'Unsupported operation: Illegal character in path: https:',
       ));
     }, overrides: <Type, Generator>{
-      FileSystem: () {
-        final FileSystem result = MemoryFileSystem(
-          style: FileSystemStyle.windows,
-        );
-        result.currentDirectory = result.systemTempDirectory.createTempSync('flutter_asset_bundle_test.');
-        return result;
-      },
+      FileSystem: () => testFileSystem,
```

Comment:
So this code in the setUp:

```
      testFileSystem = MemoryFileSystem(
        style: globals.platform.isWindows
          ? FileSystemStyle.windows
          : FileSystemStyle.posix,
      );
```

We shouldn't do. These tests should be hermetic, not relying on the host operating system.

That is, if we really want to test both Windows and posix file system styles, we should have two tests, one for windows, one for posix.

Sorry, but can you refactor this test library to not use this pattern?

Reply:
Addressed and accepted in https://github.com/flutter/flutter/pull/141657